src/components/EditPartnership/EditPartnership.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Button } from '@material-ui/core';
+import React, { FC } from 'react';
+import { Except } from 'type-fest';
+import { PartnershipType, UpdatePartnership } from '../../api';
+import { DialogForm, DialogFormProps } from '../Dialog/DialogForm';
+import { DialogState } from '../Dialog/useDialog';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../form';
+import { PartnershipCardFragment } from '../PartnershipCard/PartnershipCard.generated';
+import {
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+interface EditPartnershipProps {
+  state: DialogState;
+  partnership: PartnershipCardFragment;
+}
+
+interface DialogInput {
+  partnership: Except<UpdatePartnership, 'id'>;
+}
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  state,
+  partnership,
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const submit: DialogFormProps<DialogInput>['onSubmit'] = async (input) => {
+    const inputWithId = {
+      partnership: { ...input.partnership, id: partnership.id },
+    };
+    await updatePartnership({
+      variables: { input: inputWithId },
+      refetchQueries: ['ProjectPartnerships'],
+    });
+  };
+
+  const onDeleteClick = async () => {
```

Comment:
Hmm this is a new use case. We need to show this delete in progress state in the UI. This happens automatically with the form submit handler, but not with this handler. Let me think about the best way to handle this.

Reply:
got it